roles/validate_inventory/tasks/pxe.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Assert 'dns host' and 'tftp host' is have IPs and that is the same
+  assert:
+    that:
+      - hostvars['dns_host']['ansible_host'] is defined
+      - hostvars['tftp_host']['ansible_host'] is defined
+      - hostvars['dns_host']['ansible_host'] == hostvars['tftp_host']['ansible_host']
+    quiet: true
+  when: hostvars['dns_host']['use_pxe'] | default(false)
+
+- name: "Check if ipmitool installed"
+  ansible.builtin.shell:
+    cmd: "ipmitool -V"
+  register: ipmitool_check
+  ignore_errors: True
+  changed_when: False
+
+- name: Record failures
+  fail:
+    msg: "ipmitool must be installed for pxe installatio"
+  when: ipmitool_check.rc != 0
+
+- name: "Check if pyghmi installed"
+  ansible.builtin.shell:
+    cmd: "pip3 show pyghmi"
```

Comment:
@nocturnalastro Ansible modules are idempotent (usually), which means that if you tell Ansible to configure something that is already configured it won't do anything. 
Why use the shell module instead of the pip module for making sure the required packages are installed? it is always better to use modules instead of shell commands when possible

Reply:
btw there is a [module](https://docs.ansible.com/ansible/latest/collections/community/general/python_requirements_info_module.html#return-values) that does this check.
If the `not_found` value isn't empty - the package isn't installed

Reply:
We will open new PR for changing all the ensure to 3 way 
1. Check if the package install
2. Install the package (if have sudo access)
3. tell to install (doesn’t have sudo access)